rasa_core/domain.py

```diff
@@ -224,16 +224,24 @@ def input_states(self):
             self.slot_states + \
             self.prev_action_states
 
-    @staticmethod
-    def get_parsing_states(tracker):
+    def get_parsing_states(self, tracker):
         # type: (DialogueStateTracker) -> Dict[Text, float]
 
         state_dict = {}
 
-        # Set all found entities with the state value 1.0
+        # Set all found entities with the state value 1.0, unless they should
+        # be ignored for the current intent and are not associated with a
+        # featurized slot
         for entity in tracker.latest_message.entities:
-            key = "entity_{0}".format(entity["entity"])
-            state_dict[key] = 1.0
+            entity_name = entity.get("entity")
+            should_ignore_entity = (tracker.latest_message.intent.get("name")
```

Comment:
can we name this in a positive way and invert it, bools are easier to read if their meaning is positive.

`should_use_entity = (tracker.latest_message.intent.get("name") not in self.intents_ignore_entities)`